### PR TITLE
Feature/document commands

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

File: assets/dev/js/editor/document/history/component.js [hunk 1]
```diff
@@ -143,104 +146,76 @@ export default class Component extends elementorModules.common.Component {
...
-		}
+		argsObject.requireArgumentType( 'title', 'string' );
 
 		return elementor.history.history.startItem( args );
 	}
 
-	deleteLog( args ) {
-		// TODO: If it uses only args.id then args.id should be passed directly without args object.
-		elementor.history.history.deleteItem( args.id );
+	deleteLog( historyId ) {
```
Comment (matipojo) [inline assets/dev/js/editor/document/history/component.js]: Revert

File: assets/dev/js/editor/document/history/component.js [hunk 2]
```diff
@@ -143,104 +146,76 @@ export default class Component extends elementorModules.common.Component {
...
+		if ( historyId ) {
+			elementor.history.history.deleteItem( historyId );
+		}
 	}
 
-	endLog( args ) {
-		// TODO: If it uses only args.id then args.id should be passed directly without args object.
-		if ( args.id ) {
-			elementor.history.history.endItem( args.id );
+	endLog( historyId ) {
```
Comment (matipojo) [inline assets/dev/js/editor/document/history/component.js]: Revert

File: assets/dev/js/editor/views/add-section/base.js [hunk 3]
```diff
@@ -167,7 +167,7 @@ class AddSectionBase extends Marionette.ItemView {
 			},
 		} ).view.addElementFromPanel();
 
-		$e.run( 'document/history/end-log', { id: historyId } );
```
Comment (matipojo) [inline assets/dev/js/editor/views/add-section/base.js]: Revert

File: core/common/assets/js/components/commands.js [hunk 4]
```diff
@@ -107,19 +107,19 @@ export default class extends elementorModules.Module {
 
 		this.trigger( 'run', component, command, args );
 
-		if ( args.onBefore ) {
+		if ( args && args.onBefore ) {
```
Comment (matipojo) [inline core/common/assets/js/components/commands.js]: ?

File: core/common/assets/js/components/commands.js [hunk 5]
```diff
@@ -107,19 +107,19 @@ export default class extends elementorModules.Module {
...
 			args.onBefore.apply( component, [ args ] );
 		}
 
 		const results = this.commands[ command ].apply( component, [ args ] );
 
-		if ( args.onAfter ) {
+		if ( args && args.onAfter ) {
 			args.onAfter.apply( component, [ args ] );
```
Comment (matipojo) [inline core/common/assets/js/components/commands.js]: ?

File: core/common/assets/js/components/commands.js [hunk 6]
```diff
@@ -107,19 +107,19 @@ export default class extends elementorModules.Module {
...
 		}
 
 		this.afterRun( command, args );
 
-		if ( false === args.returnValue ) {
+		if ( args && false === args.returnValue ) {
```
Comment (matipojo) [inline core/common/assets/js/components/commands.js]: ?

File: tests/qunit/core/editor/document/commands/base/history.spec.js [hunk 7]
```diff
@@ -46,7 +46,7 @@ jQuery( () => {
 			$e.run = orig;
 
 			assert.equal( tempCommand, 'document/history/delete-log' );
-			assert.equal( tempArgs.id, instance.historyId );
```
Comment (matipojo) [inline tests/qunit/core/editor/document/commands/base/history.spec.js]: Revert

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/controls/base.js [hunk 8]
```diff
@@ -67,10 +63,15 @@ ControlBaseView = Marionette.CompositeView.extend( {
...
 					} else {
+						if ( ! settingsModel.id ) {
+							settingsModel.id = 'bc-' + elementor.helpers.getUniqueID();
+						}
+
 						// Document/General/Other control.
 						options.container = new elementorModules.editor.Container( {
 							type: 'bc-container',
 							id: settingsModel.id,
+							model: settingsModel,
```
[the remark this answers is not in the document]
Comment (iNewLegend) [reply assets/dev/js/editor/controls/base.js]: tested with essential addons

**Review by matipojo — APPROVED** [review]

**Pull request merged**